README.md

```diff
@@ -443,6 +443,27 @@ A Map parameter can be annotated with `QueryMap` to construct a query that uses
 V find(@QueryMap Map<String, Object> queryMap);
 ```
 
+This may also be used to generate the query parameters from a POJO object.
+
+```java
+@RequestLine("GET /find")
+V find(@QueryMap CustomPojo customPojo);
+```
+
+When used in this manner, the query map will be generated using member variable names as query parameter names. The following POJO will generate query params of "/find?name={name}&number={number}" (as usual, if any value is null, it will be left out).
+
+```java
+public class CustomPojo {
+  private final String name;
+  private final int number;
+
+  public CustomPojo (String name, int number) {
+    this.name = name;
+    this.number = number;
+  }
+}
+```
```

Comment:
nice docs.

We should discuss with the wider audience if we want all member variables (including non-exposed private ones), or if we want those that are exposed in some way (public or have a getter), and if we handle transient ones or not. It's worth asking as I see below in the code that we're changing access modifiers on the fly (making accessible if not) - something that may not be something we do out-of-the box or so...

Reply:
In another PR of a similar nature, I made the suggestion that we should support objects that via the Java Beans api.  That should allow for the widest range of support and interoperability.  However straight reflection is ok too from my perspective.

Reply:
I'd still argue in favor of annotating the methods/fields that you want to use with something like `@Param`. The only reason to use all fields, was for simplicity (I guess additional annotations weren't simple enough). Personally, I'm less a fan of parsing all getter method names (javabeans style) than I am of reading all fields. But if it'll get this stuff merged...

Reply:
@rage-shadowman let's get more thoughts on it from a few people. It may be that the concensus is this works just fine and we like it (and then just leave it alone). It may also be that we go Java Beans style or something else. let's see what others think before taking any drastic decision.

Reply:
If the map builder has an interface that can be passed into the feign builder, then I could even write my own in my local client that would use my annotations without complicating anything internal to feign. We could have an EJB implementation that parses getters as a separate possibility.

Reply:
I would probably call it a `QueryMapBuilder` or `ObjectQueryMapBuilder` (to differentiate it from other serializers, such as the method param serializer `Expander`, or the body param serializer `Encoder`), but I like this idea. It could just have a `Map<String, Object> toQueryMap(Object)` method (or whatever better name you guys can come up with).

Reply:
I agree that a Serializer would fit nicely, so as long as we have a sensible default Serializer.

Reply:
I created a `QueryMapEncoder` that can be specified via the `Feign.Builder`. I think it answers this concern and cleans up the naming issues.

I haven't pushed it into this pull request as I don't want to jump the gun on this. You can check that out in https://github.com/rage-shadowman/feign/tree/feature/shadowman/query-param-encoder

Should I push that into this pull request?

Reply:
@rage-shadowman I'd add your encoder to the PR and suggest that it be used by default, without any additional customization to start.

Reply:
Done.